plutus-core/untyped-plutus-core/Language/UntypedPlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -103,9 +109,10 @@ data CekValue uni fun =
 type CekValEnv uni fun = UniqueMap TermUnique (CekValue uni fun)
 
 -- | The environment the CEK machine runs in.
-data CekEnv uni fun = CekEnv
+data CekEnv uni fun s = CekEnv
     { cekEnvRuntime    :: BuiltinsRuntime fun (CekValue uni fun)
     , cekEnvBudgetMode :: ExBudgetMode
+    , cekEnvMayEmitRef :: Maybe (STRef s (DList String))
```

Comment:
again comment

Reply:
TBH this also needs a comment about which way around the logs go. It looks like you're using `snoc` so the answer is "latest logs last".